application/models/Survey.php

```diff
@@ -2286,6 +2289,56 @@ function ($s) {
         return $aSurveys;
     }
 
+    /**
+     * Returns the survey URL with the specified params.
+     * If $preferShortUrl is true (default), and an alias is available, it returns the short
+     * version of the URL.
+     * @param string|null $language
+     * @param array<string,mixed> $params   Optional parameters to include in the URL.
+     * @param bool $preferShortUrl  If true, tries to return the short URL instead of the traditional one.
+     */
+    public function getSurveyUrl($language = null, $params = [], $preferShortUrl = true)
```

Comment:
Another candidate for a unit-test, perhaps?

Reply:
Yes, also called from the functional test